schema/snapcraft.yaml

```diff
@@ -134,6 +138,10 @@ properties:
             items:
               type: string
               pattern: "^[a-zA-Z0-9][-_.a-zA-Z0-9]*$"
+          environment:
+            type: object
+            description: environment entries for the specific app.
+            minItems: 1
```

Comment:
We should probably catch empty values here as well. You might consider putting this into definitions so as to not duplicate.